src/Marten.Testing/CoreFunctionality/ejecting_a_document_type.cs

```diff
@@ -0,0 +1,117 @@
+using Marten.Testing.Documents;
+using Marten.Testing.Harness;
+using Shouldly;
+using Xunit;
+
+namespace Marten.Testing.CoreFunctionality
+{
+    public class ejecting_a_document_type : IntegrationContext
+    {
+        [Fact]
+        public void eject_a_document_type_clears_it_from_the_identity_map_regular()
```

Comment:
You probably want tests for Insert() and Update() as well as Store(). And you need to add operations from some sort of document besides `Target` to ensure that only the expected document type is impacted.

Reply:
I'm in the process of adding tests for `Update()`, but I don't understand why this test fails (even with `EjectAllOfType` commented out). I have never used the `DirtyTrackedSession` myself, so maybe I've misunderstood it, but shouldn't it track pending changes without needing to call `Update`?

```
[Fact]
public void eject_a_document_type_clears_updates_from_the_unit_of_work_dirty()
{
    var target1 = new Target { Number = 1 };
    var target2 = new Target { Number = 2 };
    var user1 = new User { Age = 10 };
    var user2 = new User { Age = 20 };

    using (var session = theStore.OpenSession())
    {
        session.Store(target1, target2);
        session.Store(user1, user2);

        session.SaveChanges();
    }

    using (var session = theStore.DirtyTrackedSession())
    {
        target1.Number = 3;
        target2.Number = 4;
        user1.Age = 30;
        user2.Age = 40;

        //session.EjectAllOfType(typeof(Target));

        session.SaveChanges();
    }

    using (var session = theStore.QuerySession())
    {
        session.Load<Target>(target1.Id).ShouldNotBeNull().Number.ShouldBe(1);
        session.Load<Target>(target2.Id).ShouldNotBeNull().Number.ShouldBe(2);
        session.Load<User>(user1.Id).ShouldNotBeNull().Age.ShouldBe(30);
        session.Load<User>(user2.Id).ShouldNotBeNull().Age.ShouldBe(40);
    }
}
```

Reply:
Just what you said, the dirty tracking checks all documents that were loaded through the session at the time SaveChanges() was called. Very inefficient, only useful as a convenience when perf doesn't matter much.

Reply:
Ok. So why doesn't the test above work? When you do e.g. `target1.Number = 3` inside the `DirtyTrackedSession` and then `SaveChanges`, why is that change not committed?

Reply:
I just realized something myself.. Do you need to load the objects from the actual `DirtyTrackedSession` for it to know about them? That would probably make sense.

Reply:
Yep, that was it. See latest commit.